src/main/java/gov/nysenate/openleg/config/OpenLegComboPooledDataSource.java

```
@@ -0,0 +1,44 @@
+package gov.nysenate.openleg.config;
+
+import com.mchange.v2.c3p0.AbstractComboPooledDataSource;
+
+import javax.naming.Referenceable;
+import java.io.*;
+
+/**
+ * This class is copied nearly exactly from ComboPooledDataSource, except with an overridden close method.
+ */
+final class OpenLegComboPooledDataSource extends AbstractComboPooledDataSource implements Serializable, Referenceable {
+    public OpenLegComboPooledDataSource() {
+        super();
+    }
+
+    @Serial
+    private static final long serialVersionUID = 1;
+    private static final short VERSION = 0x0002;
+
+    @Serial
+    private void writeObject(ObjectOutputStream oos) throws IOException {
+        oos.writeShort(VERSION);
+    }
+
+    @Serial
+    private void readObject(ObjectInputStream ois) throws IOException, ClassNotFoundException {
+        short version = ois.readShort();
+        if (version != VERSION)
+            throw new IOException("Unsupported Serialized Version: " + version);
+    }
+
+    /**
+     * Intercepts an error thrown deep within the stack trace at shutdown.
+     * This error warns of a memory leak, but since we're shutting down,
+     * it can be safely ignored.
+     */
+    @Override
+    public void close() {
+        try {
+            super.close();
+        }
+        catch (Exception ignored) {}
+    }
```

This doesn't just hide memory leak errors, it will hide all errors from a connection being closed right? Personally, I would prefer to see a few erroneous errors on shutdown than to have all errors hidden. Perhaps  @kzalewski has an opinion?

---

That's right, I changed this section to more directly address the issue instead.